mlrun/datastore/sources.py

```diff
@@ -329,10 +329,6 @@ def __init__(
             raise mlrun.errors.MLRunInvalidArgumentError(
                 "cannot specify both table and query args"
             )
-        if not query and not table:
-            raise mlrun.errors.MLRunInvalidArgumentError(
-                "must specify at least one of table or query args"
-            )
```

Comment:
It's a bit unfortunate we cannot keep this check here - this enables checking on client-side whether required parameters exist. Otherwise when processing remotely and creating this source without both `query` and `table` you'll only get the error much later in the process, when ingesting. 
Still, I understand why this cannot be here, but I wonder if there's no way to keep this check on client side while avoiding it when doing `from_dict`.

Reply:
Agreed that there's no obvious way to do this without significant refactoring.